stdlib/http/src/main/ballerina/src/http/http_commons.bal

```diff
@@ -242,10 +242,10 @@ function buildResponse(ResponseMessage message) returns Response {
         response.setTextPayload(message);
     } else if (message is xml) {
         response.setXmlPayload(message);
-    } else if (message is json) {
-        response.setJsonPayload(message);
     } else if (message is byte[]) {
         response.setBinaryPayload(message);
+    } else if (message is json) {
+        response.setJsonPayload(message);
```

Comment:
What is the reason to interchange this `else if` condition?

Reply:
Take the following code snippet

```ballerina
type Person record {
    
};

public function main() {
   int[] | json | Person value = "";
   if (value is json) {
       // do something
   } else if( value is int[]) { // <--- error on this line
       //do something
   }
}
```
The above code snippet prints out the following error. 
`Person' will not be matched to 'int[]'`

This is because the `is json` check also captures `int[]`, since `int` can be used as `json`. 
And when we are in the `else` clause the type is narrowed to `Person`. 

The reason for interchanging `else if` above is because of this but for type `byte[]` 

ps: please correct me if my understanding is wrong here @MaryamZi  @pubudu91 @rdhananjaya  :)

Reply:
Yeah, basically a `byte` is `json`, and therefore a `byte[]` is also `json`.

So, when using the type guard, if the `json` type test comes before the `byte[]` type test, the `byte[]` type test will never be evaluated to true since any `byte[]` is a `json`.